backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -919,13 +922,8 @@ private fun ObjCExportCodeGenerator.createTypeAdapter(
 
         } else {
             // Mark it as non-overridable:
-            baseMethods.distinctBy { namer.getSelector(it) }.forEach { base ->
-                reverseAdapters += KotlinToObjCMethodAdapter(
-                        namer.getSelector(base),
-                        -1,
-                        -1,
-                        NullPointer(int8Type)
-                )
+            baseMethods.distinctBy { namer.getSelector(it) }.forEach { baseMethod ->
```

Comment:
This is wrong.
The following test should fail:
```
// Kotlin
open class A {
    fun foo() {}
}

// Swift
class B : A {
    override func foo() {}
}

B()
```

However with this patch it doesn't fail.

Reply:
Why shall it fail?

Reply:
It seems, it fails in the runtime with ` 'NSGenericException', reason: '[op.B foo] can't be implemented'` while better fail in the compile time. Anyway, restored old behavior.

Reply:
Unfortunately, this doesn't seem to be possible in the compile time as Objective-C doesn't have a concept of `final` methods.